rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -375,6 +375,13 @@ public Optional<Federation> getProposedFederation() {
         return provider.getProposedFederation(constants, activations);
     }
 
+    @Override
+    public int getProposedFederationSize() {
+        return getProposedFederation()
+            .map(Federation::getSize)
+            .orElse(FederationChangeResponseCode.PROPOSED_FEDERATION_NON_EXISTENT.getCode());
```

Comment:
I know this is how it's handled with the retiring federation. But I wonder, should we perhaps return an Optional here and then et the Bridge decide to return `-1` when there is no proposed federation?

I feel like it's the Bridge responsibility to decide on the default value. For any other internal uses of this method, an Optional return might be more useful

Reply:
Seems like `getRetiringFederationCreationTime` is handled that way

Reply:
Yes, I agree. I was mainly just imitating how retiring federation does it. I have updated. Hopefully, we can do this for the rest of the methods and let the Bridge decide the defaults.